protostar/protostar_toml/protostar_toml_version_checker.py

```diff
@@ -0,0 +1,48 @@
+from typing import Any
+
+from protostar.protostar_exception import ProtostarException
+from protostar.protostar_toml.io.protostar_toml_reader import ProtostarTOMLReader
+from protostar.utils import VersionManager
+
+
+class ProtostarTOMLVersionChecker:
+    SKIP_FOR = ["init", "upgrade"]
+
+    def __init__(
+        self,
+        protostar_toml_reader: ProtostarTOMLReader,
+        version_manager: VersionManager,
+    ):
+        self._protostar_toml_reader = protostar_toml_reader
+        self._version_manager = version_manager
+
+    def run(self, args: Any):
+        if args.command in self.SKIP_FOR:
+            return
+
+        declared_version_str = self._protostar_toml_reader.get_attribute(
+            "config", "protostar_version"
+        )
+
+        assert (
```

Comment:
Can we use ProtostarException instead of assert? 

I know it is not supposed to happen but it is kinda user input validation, I think oldest version of protostar don't have this field

Reply:
This is going to happen when we remove the config section. The error here should tell the user to upgrade the Protostar.